openhexa/sdk/pipelines/runtime.py

```diff
@@ -19,6 +22,47 @@ def import_pipeline(pipeline_dir_path: str):
     return pipeline
 
 
+def get_pipeline_specs(pipeline_dir_path) -> typing.Tuple[PipelineSpecs, typing.List[PipelineParameterSpecs]]:
+    pipeline_dir = os.path.abspath(pipeline_dir_path)
+    sys.path.append(pipeline_dir)
+    pipeline_node = None
+
+    with open(pipeline_dir_path / Path("pipeline.py")) as f:
+        tree = ast.parse(f.read())
+        for node in ast.walk(tree):
+            if isinstance(node, ast.FunctionDef) and [
+                decorator for decorator in node.decorator_list if decorator.func and decorator.func.id == "pipeline"
+            ]:
+                pipeline_node = node
+                break
+
+    if pipeline_node:
+        pipeline_decorator = [
```

Comment:
This second list comprehension feels weird. I have the impression that there is an easier way to get the pipeline node and the decorator in one pass.

Reply:
Get the pipeline node and parameter decorators?